gw_eccentricity/exceptions.py

```diff
@@ -0,0 +1,58 @@
+"""Custom exception classes for gw_eccentricity."""
+
+
+class InsufficientExtrema(Exception):
```

Comment:
I don't get why you need this so far. Please explain in the comments why you need this, for the sake of future you, if nothing else.

Reply:
This is to help catching exception that is solely due to insufficient extrema because generic exception can happen for many things. This could be useful when you use try except kind of statement where you can specify what kind of exception one is looking at. I have added more documentation.

Reply:
I still don't get why we need this.
Isn't the idea like:
If the new flag is passed, ecc=meanano=0 is returned when appropriate.
Why does the user or a postprocessing script need to worry about handling different types of exceptions?

Reply:
I think I get your point that if you avoid the extrema-related exception using the set_failures_to_zero option then probably there is no advantage in adding this custom exception. My original thinking was that irrespective of the set_failures_to_zero, the user will have a clear idea of what kinds of exception it is and even if they did not set the set_failures_to_zero, they will have the option to do whatever they want to do in such cases.

Having said that if you think it is not adding anything, I can go back to the generic exception.

Reply:
Yeah, I think get rid of it. Might lead to confusion while solving a problem that's not a real problem. Same for the other thing in the exceptions.py file?

Reply:
Done!